packages/ag-charts-community/src/util/json.ts

```diff
@@ -111,26 +117,31 @@ export function shallowClone<T>(source: T): T {
  *
  * @param json to traverse
  * @param visit callback for each non-primitive and non-array object found
- * @param opts
- * @param opts.skip property names to skip when walking
- * @param jsons to traverse in parallel
+ * @param skip property names to skip when walking
+ * @param parallelJson to traverse in parallel
  */
-export function jsonWalk<T>(json: T, visit: (...nodes: T[]) => void, opts?: { skip?: string[] }, ...jsons: T[]) {
+export function jsonWalk<T, C>(
+    json: T,
+    visit: (node: T, parallelNode?: T, ctx?: C) => void,
+    skip?: Set<string>,
+    parallelJson?: T,
+    ctx?: C
+) {
     if (isArray(json)) {
-        visit(json, ...jsons);
-        json.forEach((node, index) => {
-            jsonWalk(node, visit, opts, ...keyMapper(jsons, index));
-        });
+        visit(json, parallelJson, ctx);
+        let index = 0;
+        for (const node of json) {
+            jsonWalk(node, visit, skip, (parallelJson as any[])?.[index], ctx);
+            index++;
+        }
     } else if (isPlainObject(json)) {
-        visit(json, ...jsons);
+        visit(json, parallelJson, ctx);
```

Comment:
Why do we need `ctx` here?

Reply:
This is for the `ChartTheme.templateTheme()` case where we need to pass around the map of values to apply - so we need a common 'context' that gets passed in to every `visit()` callback.